app/controllers/collection_associations_controller.rb

```diff
@@ -0,0 +1,60 @@
+# :nodoc:
+class CollectionAssociationsController < CmrSearchController
+  include ChooserEndpoints
+
+  before_action :set_variable
+  before_action :add_high_level_breadcrumbs
+
+  def index
+    # Default the page to 1
+    page = params.fetch('page', 1)
+
+    association_response = cmr_client.get_variables({ concept_id: params[:variable_id] }, token)
+
+    association_list = if association_response.success?
+                         # Retreive the service options and sort by name, ignoring case
+                         concept_ids = association_response.body.fetch('items', []).first.fetch('associations', {}).fetch('collections', []).map { |collection| collection['concept-id'] }
+
+                         concept_ids.any? ? get_provider_collections(concept_id: concept_ids, sort_key: 'entry-title').fetch('items', []) : []
+                       else
+                         []
+                       end
+
+    @associations = Kaminari.paginate_array(association_list, total_count: association_list.count).page(page).per(RESULTS_PER_PAGE)
+  end
+
+  def new
+    add_breadcrumb 'New', new_variable_collection_association_path(params[:variable_id])
+
+    super
+
+    association_response = cmr_client.get_variables({ concept_id: params[:variable_id] }, token)
+
+    @previously_associated_collections = association_response.body.fetch('items', []).first.fetch('associations', {}).fetch('collections', []).map { |collection| collection['concept-id'] }
```

Comment:
You're parsing the association concept ids multiple times - is it worth making it into a method?